### PR TITLE
Use TFS to mark question attempts

### PR DESCRIPTION
**Pull request by jmcgettrick**

### CODE REVIEW

File: classes/modules/turnitin_quiz.class.php [hunk 1]
```diff
@@ -61,4 +63,47 @@ public function get_current_gradequery($userid, $moduleid, $itemid = 0) {
...
+    }
+
+    // Set a new mark for a question attempt based on the grade given for the answer in TFS.
+    public function update_mark($attemptid, $identifier, $userid, $grade, $quizgrade) {
+        global $DB;
+
+        $transaction = $DB->start_delegated_transaction();
+
+        $attempt = quiz_attempt::create($attemptid);
+        $quba = question_engine::load_questions_usage_by_activity($attemptid);
```
Comment (dwinn) [inline classes/modules/turnitin_quiz.class.php:79]: What is $quba? Might be better to give a more descriptive name. :)

Comment (jmcgettrick) [reply classes/modules/turnitin_quiz.class.php:79]: `questions_usage_by_activity` - it's called that all through the quiz code. It helped me understand it easier by leaving it as that.

Comment (dwinn) [reply classes/modules/turnitin_quiz.class.php:79]: Fair point. I wonder why Moodle called it that.

**Comment by jmcgettrick** [comment]
Rebased

**Pull request merged**